modules/home-manager/zsh-syntax-highlighting.nix

```diff
@@ -2,14 +2,22 @@
 let
   inherit (config.catppuccin) sources;
   inherit (lib) ctp;
-  cfg = config.programs.zsh.syntaxHighlighting.catppuccin;
+  cfg = config.catppuccin.zsh-syntax-highlighting;
   enable = cfg.enable && config.programs.zsh.syntaxHighlighting.enable;
 in
 {
-  options.programs.zsh.syntaxHighlighting.catppuccin = ctp.mkCatppuccinOpt {
+  options.catppuccin.zsh-syntax-highlighting = ctp.mkCatppuccinOpt {
     name = "Zsh Syntax Highlighting";
   };
 
+  # FIXME: this leads to an error
+  # imports = lib.ctp.mkCatppuccinRenamedOpt [
+  #   "programs"
+  #   "zsh"
+  #   "syntaxHighlighting"
+  #   "catppuccin"
+  # ] "zsh-syntax-highlighting";
```

Comment:
What is the error here? 🙀

Reply:
It results in this
> error: evaluation aborted with the following error message: 'cannot find attribute `programs.zsh.syntaxHighlighting.catppuccin.flavor''

Reply:
My guess is that `programs.zsh.syntaxHighlighting` is defined as a submodule. I tried to define renamed options in `programs.jq.colors` which is also a submodule and got the same error

Reply:
Yeah seems so, https://github.com/nix-community/home-manager/blob/67cd4814a247fd0fe97171acb90659f7e304bcb8/modules/programs/zsh.nix#L238.

How would we get around this? Perhaps leaving the old options and including a warning?

Reply:
Yes, try to manually define the renamed options and the warning and set the new options to the old ones by default. Basically replicating `mkRenamedOptionModule`

Reply:
addressed https://github.com/catppuccin/nix/compare/4c7f5b65973d76448834398f221692cf54ca24b9..0c1924eb253c0c0eb15178e24b52d5fb285982fc

Reply:
Did you try to set the new options to the old ones by default?